packages/nextjs/src/edge/wrapApiHandlerWithSentry.ts

```diff
@@ -10,22 +10,24 @@ export function wrapApiHandlerWithSentry<H extends EdgeRouteHandler>(
   handler: H,
   parameterizedRoute: string,
 ): (...params: Parameters<H>) => Promise<ReturnType<H>> {
-  return async function (this: unknown, ...args: Parameters<H>): Promise<ReturnType<H>> {
-    const req = args[0];
+  return new Proxy(handler, {
+    apply: async (wrappingTarget, thisArg, args: Parameters<H>) => {
```

Comment:
Is this supported, e.g. having an async `apply`? Just wondering :D I guess we don't even really need it here, though? Or am I missing something?

Reply:
Not sure about the term "supported" but from my testing, using an async function just seems to return a promise which is about what I would expect to happen. We definitely need it here though because we wanna await the result so that the events are flushed.

Reply:
Ah, good, if it works then 👍 maybe it is just a restriction for `get` based proxies, I vaguely remember having issues with such a thing in the past. But all good then, nice!